src/rred_reports/masterfile.py

```diff
@@ -118,18 +119,20 @@ def parse_masterfile(file: Path) -> dict[str, pd.DataFrame]:
         dict[str, pd.DataFrame]: Dictionary of dataframes
     """
     full_data = pd.read_excel(file)
+    # drop out teachers at this point, so that if a teacher has changed title between years, we take the most recent one
+    filtered_data = full_data[~full_data["reg_rr_title"].isin(["Teacher Leader", "Teacher Leader in Training"])].copy()
 
-    def clmnlist(i, data=full_data):
+    def clmnlist(i, data=filtered_data):
```

Comment:
Missing type annotations here
```suggestion
    def clmnlist(i: int, data: pd.DataFrame=filtered_data) -> list:
```